apiextensions/storageversion/migrator.go

```diff
@@ -62,6 +62,11 @@ func (m *Migrator) Migrate(ctx context.Context, gr schema.GroupResource) error {
 		return fmt.Errorf("unable to fetch crd %s - %w", gr, err)
 	}
 
+	// don't migrate storage version if CRD has a single storage in its status
+	if len(crd.Status.StoredVersions) == 1 {
+		return nil
+	}
+
```

Comment:
I think we need a stronger check and assert the storage version in the status is length 1 but it also is the version that's expected in the spec

Reply:
The reason why I suggest this is because I believe when you switch the storage version in the spec it won't show up in your status until certain resources are being stored using that version - that only happens when new resources are created and older ones and updated

Reply:
Got it. I've updated the check to ensure that the version in the status and spec are the same. 

I do have a question about the tests though. If the CRD doesn't need to migrate versions, I just return. What is the best way to test this behavior? Currently I'm checking that Migrate didn't fail and that the storage version is correct, but that's a bit icky. Should I be returning an errors instead if the CRD version doesn't need to migrate?

Reply:
K8s has a fake client that we are using in the test you could

1. Go through all the actions and ensure there are no patches
https://github.com/kubernetes/client-go/blob/bc7002f524b2527ceede8092e3e5232c005a42fb/testing/fake.go#L214

2. Prepend a Reactor and when there's a patch fail the test
https://github.com/kubernetes/client-go/blob/bc7002f524b2527ceede8092e3e5232c005a42fb/testing/fake.go#L103 

The first might be more straight forward

Reply:
Thank you!